flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1051,6 +1051,14 @@ private SqlNode validateScopedExpression(SqlNode topNode, SqlValidatorScope scop
         if (outermostNode.isA(SqlKind.TOP_LEVEL)) {
             registerQuery(scope, null, outermostNode, outermostNode, null, false);
         }
+        // ----- FLINK MODIFICATION BEGIN -----
+        if (outermostNode instanceof SqlSelect) {
+            SqlSelect select = (SqlSelect) outermostNode;
```

Comment:
I am not an expert on the table planner - but had some basic questions. 
1) I am not seeing how we know it is a select * . Is this related to `if (outermostNode instanceof SqlSelect) {`
2) Will this effect all `select *` code , include the working non UDF `select *`'s? If so is there now redundant code we need to remove that dealt with the * later?

Reply:
Thanks you @davidradl for your review!

1. I am not isolating `select *` for just the `select` node. This is about the star expansion in function calls only
2. This would not expand `*` alone, but only in `SqlCall`. The implementation of that is quite hardcoded in the `SqlValidator` and I did not touch that. However, I am able to re-use the same `expandStar` method.

Does this answer your questions?